### PR TITLE
images/capi/packer/azure: small SIG related improvements and fixes

### PR DESCRIPTION
**Pull request by invidian**
What this PR does / why we need it:

See commit messages for more details.

**Additional context**

Identified while working on https://github.com/flatcar-linux/Flatcar/issues/734.

### CODE REVIEW

File: images/capi/packer/azure/packer.json [hunk 1]
```diff
@@ -61,13 +61,12 @@
         "plan_publisher": "{{user `plan_image_publisher`}}"
       },
       "private_virtual_network_with_public_ip": "{{user `private_virtual_network_with_public_ip`}}",
+      "shared_gallery_image_version_exclude_from_latest": "{{ user `exclude_from_latest` }}",
```
Comment (kkeshavamurthy) [inline images/capi/packer/azure/packer.json:64]: Can this be added to `packer-windows` as well?

File: images/capi/packer/azure/scripts/init-sig.sh [hunk 2]
```diff
@@ -34,38 +34,38 @@ SIG_TARGET=$1
 
 case ${SIG_TARGET} in
   ubuntu-1804)
-    create_image_definition ${SIG_TARGET} "18.04-LTS" "V1" "Linux"
+    create_image_definition capi-${SIG_TARGET} "18.04-LTS" "V1" "Linux"
```
[the remark this answers is not in the document]
Comment (kkeshavamurthy) [reply images/capi/packer/azure/scripts/init-sig.sh:37]: What is the reason for adding a `capi` prefix to the image-definition? This hardcoding make it difficult to override it.

[the remark this answers is not in the document]
Comment (invidian) [reply images/capi/packer/azure/scripts/init-sig.sh:37]: Hm, so this actually moves the definition one level up the stack (from hardcoded in function to function argument), so it makes it easier to override for a single target, though still not actually possible. I'll try to figure how to proceed with this to actually make it configurable when needed.

[the remark this answers is not in the document]
Comment (invidian) [reply images/capi/packer/azure/scripts/init-sig.sh]: I dropped this patch until I have full solution.

File: images/capi/packer/azure/packer.json (hunk 1 above)
Comment (invidian) [reply images/capi/packer/azure/packer.json:64]: Done. Sorry for the delay.

**Comment by kkeshavamurthy** [comment]
/lgtm
@invidian If this PR is good to go can you please squash the commits

**Comment by invidian** [comment]
> @invidian If this PR is good to go can you please squash the commits

Thanks for having a look. I've rebased the PR and it's ready to go. I wouldn't squash the commits though, as they are distinct and of good quality.

**Comment by invidian** [comment]
CI is flaky :(

**Comment by jsturtevant** [comment]
looks related to pip: 

```
 Complete output from command python setup.py egg_info:\n    Traceback (most recent call last):\n      File \"<string>\", line 1, in <module>\n      File \"/tmp/pip-build-ixrvk6wk/bcrypt/setup.py\", line 11, in <module>\n        from setuptools_rust import RustExtension\n    ModuleNotFoundError: No module named 'setuptools_rust'\n    \n            =============================DEBUG ASSISTANCE==========================\n            If you are seeing an error here please try the following to\n            successfully install cryptography:\n    \n            Upgrade to the latest pip and try again. This will fix errors for most\n            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip\n            =============================DEBUG ASSISTANCE==========================\n    \n    \n    ----------------------------------------\n\n:stderr: Command \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-ixrvk6wk/bcrypt/\n"}[0m
``` 

We've had some issue with it in the past but his looks slightly different https://github.com/kubernetes-sigs/image-builder/issues/766

/retest

**Comment by mboersma** [comment]
Now that #958 is merged, this should work after rebasing.

**Comment by invidian** [comment]
Rebased, let's see...

**Review by mboersma — COMMENTED** [review]
/lgtm

Sorry to let this dangle so long @invidian, thanks for the fixes!

**Pull request merged**